pkg/model/gcemodel/api_loadbalancer.go

```diff
@@ -101,13 +84,118 @@ func (b *APILoadBalancerBuilder) Build(c *fi.ModelBuilderContext) error {
 
 	// Allow traffic into the API (port 443) from KubernetesAPIAccess CIDRs
 	{
+		network, err := b.LinkToNetwork()
+		if err != nil {
+			return err
+		}
 		b.AddFirewallRulesTasks(c, "https-api", &gcetasks.FirewallRule{
 			Lifecycle:    b.Lifecycle,
-			Network:      b.LinkToNetwork(),
+			Network:      network,
 			SourceRanges: b.Cluster.Spec.KubernetesAPIAccess,
 			TargetTags:   []string{b.GCETagForRole(kops.InstanceGroupRoleMaster)},
 			Allowed:      []string{"tcp:443"},
 		})
 	}
 	return nil
+
+}
+
+// createInternalLB creates an internal load balancer for the cluster.  In
+// GCP this entails creating a health check, backend service, and one forwarding rule
+// per specified subnet pointing to that backend service.
+func createInternalLB(b *APILoadBalancerBuilder, c *fi.ModelBuilderContext) error {
+	lbSpec := b.Cluster.Spec.API.LoadBalancer
+	hc := &gcetasks.HealthCheck{
+		Name:      s(b.NameForHealthCheck("api")),
+		Port:      443,
+		Lifecycle: b.Lifecycle,
+	}
+	c.AddTask(hc)
+	var igms []*gcetasks.InstanceGroupManager
+	for _, ig := range b.InstanceGroups {
+		if ig.Spec.Role != "Master" {
+			continue
+		}
+		if len(ig.Spec.Zones) > 1 {
+			return fmt.Errorf("Instance group %q has %d zones, which is not yet supported for GCP.", ig.GetName(), len(ig.Spec.Zones))
+		}
+		if len(ig.Spec.Zones) == 0 {
+			return fmt.Errorf("Instance group %q must specify exactly one zone.", ig.GetName())
+		}
+		zone := ig.Spec.Zones[0]
+		igms = append(igms, &gcetasks.InstanceGroupManager{Name: s(gce.NameForInstanceGroupManager(b.Cluster, ig, zone)), Zone: s(zone)})
+	}
+	bs := &gcetasks.BackendService{
+		Name:                  s(b.NameForBackendService("api")),
+		Protocol:              s("TCP"),
+		HealthChecks:          []*gcetasks.HealthCheck{hc},
+		Lifecycle:             b.Lifecycle,
+		LoadBalancingScheme:   s("INTERNAL"),
+		InstanceGroupManagers: igms,
+	}
+	c.AddTask(bs)
+	for _, sn := range lbSpec.Subnets {
+		network, err := b.LinkToNetwork()
+		if err != nil {
+			return err
+		}
+		t := true
+		subnet := &gcetasks.Subnet{
+			Name:    s(sn.Name),
+			Network: network,
+			Shared:  &t,
+			// Override lifecycle because these subnets are specified
+			// to already exist.
+			Lifecycle: fi.LifecycleExistsAndWarnIfChanges,
+		}
+		if subnetNotSpecified(sn, b.Cluster.Spec.Subnets) {
```

Comment:
This is clever.  For a future PR, could we also automatically associate to the subnet(s) in use if no Subnets are specified?

Reply:
Added todo.